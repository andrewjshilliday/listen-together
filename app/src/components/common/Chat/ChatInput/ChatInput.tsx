import React, { useState } from 'react';
import { TextField, InputAdornment, IconButton } from '@material-ui/core';
import SendIcon from '@material-ui/icons/Send';
import { useChat } from '../../../providers';
import styled from 'styled-components';

const ChatInput: React.FC = (props: any) => {
  const chatProvider = useChat();
  const [message, setMessage] = useState('');

  const sendMessage = () => {
    chatProvider.actions.sendMessage(message);
    setMessage('');
  }

  return (
    <InputContainer>
      <TextField label="Type a message" value={message} onChange={(e) => setMessage(e.target.value)}
        onKeyPress={(e) => {
          if (e.key === 'Enter') {
            e.preventDefault();
            sendMessage();
          }
        }}
        InputProps={{
          endAdornment: (
            <InputAdornment position="end">
              <IconButton onClick={() => sendMessage()}>
                <SendIcon />
              </IconButton>
            </InputAdornment>
          )
        }} />
    </InputContainer>
  );
}

export default ChatInput;


const InputContainer = styled.div`
  height: 55px;
  margin: 0 25px;
  border-top: 1px solid #ccc;
  >div {
    width: 100%;
    >div button {
      :hover { color: var(--primary); }
    }
  }
`;
