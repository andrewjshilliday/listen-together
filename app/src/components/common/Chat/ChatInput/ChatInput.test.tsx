import React from 'react';
import renderer from 'react-test-renderer';
import ChatInput from './ChatInput';

jest.mock('../../../providers/ChatProvider', () => ({
  useChat: () => ({
    messages: [],
    actions: {
      sendMessage: jest.fn()
    }
  })
}));

it('renders correctly', () => {
  const tree = renderer.create(<ChatInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
