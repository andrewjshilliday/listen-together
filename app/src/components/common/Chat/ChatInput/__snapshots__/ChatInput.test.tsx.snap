// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm dFMjiv"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
    onKeyPress={[Function]}
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Type a message
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
